// Copyright 2020 The gRPC Authors

syntax = "proto3";

package envoy.service.load_stats.v3;

import "envoy/config/core/v3/base.proto";
import "envoy/config/endpoint/v3/load_report.proto";

import "google/protobuf/duration.proto";

service LoadReportingService {

  rpc StreamLoadStats(stream LoadStatsRequest) returns (stream LoadStatsResponse) {
  }
}

message LoadStatsRequest {

  config.core.v3.Node node = 1;

  repeated config.endpoint.v3.ClusterStats cluster_stats = 2;
}

message LoadStatsResponse {

  repeated string clusters = 1;

  bool send_all_clusters = 4;

  google.protobuf.Duration load_reporting_interval = 2;

  bool report_endpoint_granularity = 3;
}
